### PR TITLE
consumer(ticdc): update the logic of consumer.

### LINKED ISSUE #4128: TiCDC add a test to cover the invariant between `MqMessageTypeResolved` Event and other Events.
### Is your feature request related to a problem?

TiCDC would send `MqMessageTypeResolved` messages to the MQ sink, with protocol such as `open-protocol`, `craft`, and `canal-json`.

We need to find a way to guarantee the semantic of the `MqMessageTypeResolved` message:

If the downstream MQ sink receive a `MqMessageTypeResolved` message with `ts = a`, it should be guaranteed that:
* `all received messages should have ts less than a`
* `no more messages with ts less than a will be received by the downstream`.

### Describe the feature you'd like

We have `kafka-consumer` at the moment, investigating whether it can be used to solve the problem mentioned above.

### Describe alternatives you've considered

_No response_

### Teachability, Documentation, Adoption, Migration Strategy

_No response_

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #4128 

Use `kafka-consumer` program to maintain the order relationship between `commitTs` and `watermarkTs`.

`watermarkTs` also known as `ResolvedTs` in TiCDC source code.

### What is changed and how it works?

Modify the implementation detail of the `kakfa-consumer` program.

* If meet `ResolvedTs` fallback, which means receive `Resolved Event` with `Ts` less than current `ResolvedTs`, just crash the consumer.
* If meet `Row Changed Event` or `DDL Event` with `Ts` not greater than the `globalResolvedTs`, just ignore these events should already be seen.

After this PR, the consumer should still be running stable.

### Check List 

Tests 

 - No code

Code changes

Side effects

Related changes

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Comment by asddongmen** [comment]
please remove this TODO:https://github.com/pingcap/tiflow/blob/master/tests/integration_tests/move_table/run.sh#:~:text=%23%20TODO%3A%20remove%20after%20kafka%2Dconsumer/main.go%20is%20fixed 
and try to run kafka integration test

File: cmd/kafka-consumer/main.go [hunk 1]
```diff
@@ -453,11 +453,18 @@ ClaimMessages:
 					log.Fatal("decode message value failed", zap.ByteString("value", message.Value))
 				}
 				resolvedTs := atomic.LoadUint64(&sink.resolvedTs)
-				if resolvedTs < ts {
+				if ts < resolvedTs {
+					log.Fatal("partition resolved ts fallback",
```
Comment (overvenus) [inline cmd/kafka-consumer/main.go:457]:
```suggestion
					log.Panic("partition resolved ts fallback",
```

Prefer `log.Panic`.

File: cmd/kafka-consumer/main.go [hunk 2]
```diff
@@ -440,31 +441,37 @@ ClaimMessages:
...
 				}
 			case model.MqMessageTypeResolved:
 				ts, err := batchDecoder.NextResolvedEvent()
 				if err != nil {
-					log.Fatal("decode message value failed", zap.ByteString("value", message.Value))
+					log.Panic("decode message value failed", zap.ByteString("value", message.Value))
 				}
 				resolvedTs := atomic.LoadUint64(&sink.resolvedTs)
-				if resolvedTs < ts {
+				if ts < resolvedTs {
```
Comment (overvenus) [inline cmd/kafka-consumer/main.go:456]: Could you add some comments about the ordering of the ts and the resolvedTs?

**Review by overvenus — APPROVED** [review]
LGTM

**Review by asddongmen — APPROVED** [review]

**Pull request merged**